File: pset6/readability/readability.py
from cs50 import get_string

string_name = get_string("Text: ")# get string from user:
l = 0
g = 0 
w = 0
for char in string_name: #counter for Letterand sentenc 
    
    if char.isalpha():
        l += 1
    if char == ' ':
        w += 1
    if char in ['!', '.' , '?']:
        g += 1
w +=1

total = round (0.0588 * (100 *  (l /  w) ) - 0.296 * (100 *  (g /  w) ) - 15.8) #calculate

if total < 16 and total >= 0:#print the result
    print(f"Grade {total}")
       
elif total >= 16:
    print("Grade 16+")
    
else:
    print("Before Grade 1")

File: pset2/readability/readability.c
 #include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)

{
    string s = get_string("Text: ");
     
    int l = 0;
    
    int g = 0;
    
    int w = 1;
    // counter
    for ( int i = 0 ,c = strlen(s); i < c; i++  )   
    
    {
        if (isalpha(s[i]))
        
         {
            
        l++;
        
         }
        
        else if ( s[i] == '!'|| s[i] == '.' || s[i] == '?' )
        
         {
            g++;
         }
        
      else if ( s[i]  ==  ' ' )
        
         {
             w++;
         }
    }    
    
    // clculate 
    float f = 0.0588 * (100 * (float) l / (float) w ) - 0.296 * (100 * (float) g / (float) w ) - 15.8;
    if ( f < 16 && f >= 0)
    
    //printf
    {

        printf("Grade %i\n", (int) round(f));
    }
    else if ( f >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }

}
